javascript/ql/src/semmle/javascript/Expr.qll

```diff
@@ -161,6 +161,39 @@ class Expr extends @expr, ExprOrStmt, ExprOrType, AST::ValueNode {
    * file that was extracted without type information.
    */
   Type getType() { ast_node_type(this, result) }
+
+  /**
+   * Holds if the syntactic context that the expression appears in relies on the expression
```

Comment:
Perhaps you could describe a little clearer what "relies on" means here? Is the criterion that null/undefined in this position is either converted to a number or a string, or causes a runtime error?

Reply:
This still needs addressing.